layer4/connection.go

```diff
@@ -187,8 +229,15 @@ var (
 	listenerCtxKey caddy.CtxKey = "listener"
 )
 
+const PrefetchChunkSize = 1024
```

Comment:
Does this need to be exported?

Should this be configurable instead?

Reply:
No its an implementation detail. And also just an attempt to keep the buffers smaller, since most matching will not need the full 8 KiB max buffer size. If we make something configurable then the max matching buffer size, but for now I do not think this is necessary.

Reply:
Ok cool